widgets/sites/src/SiteLocationInput.tsx

```diff
@@ -1,17 +1,23 @@
-// @ts-nocheck File not migrated fully to TS
+import type { Field } from 'app/widgets/common/types';
+import { useState, useEffect } from 'react';
+import type { SyntheticEvent } from 'react';
 import SiteLocationMap from './SiteLocationMap';
 
-export default function SiteLocationInput({ value, onChange, toolbox }) {
-    const { useState, useEffect } = React;
+export interface SiteLocationInputProps {
+    value?: string;
+    toolbox: Stage.Types.Toolbox;
+    onChange: (event: SyntheticEvent | null, field: Omit<Field, 'type' | 'checked'>) => void;
+}
 
+export default function SiteLocationInput({ value = '', onChange, toolbox }: SiteLocationInputProps) {
     const [mapOpen, setMapOpen] = useState(false);
     const [enteredValue, setEnteredValue] = useState(value || '');
 
     useEffect(() => setEnteredValue(value), [value]);
 
-    function onLocationChange(newValue) {
-        setEnteredValue(newValue);
-        onChange(null, { name: 'siteLocation', value: newValue });
+    function onLocationChange(changedLocation: string) {
+        setEnteredValue(changedLocation);
+        onChange(null, { name: 'siteLocation', value: changedLocation });
```

Comment:
Optional:

I would go with `newValue` or `newLocation` as it seems more common (than `changed*`) pattern we try to follow in this repo:

```suggestion
    function onLocationChange(newLocation: string) {
        setEnteredValue(newLocation);
        onChange(null, { name: 'siteLocation', value: newLocation });
```

Reply:
Sure, if that's more common approach, then let's go with it 💪 
Thanks btw for propagating that knowledge 🎉